### PR TITLE
Fix/various issues 221219

### PR DESCRIPTION
**Pull request by J4bbi**
- Creates a default sort property that can be set for each entity
- Fixes the empty check for columns
- Changes related objects for repositories to be a M2M field. If you think this will work fine @philipkcl  then I will proceed to do the same change for the other entities.

### CODE REVIEW

File: institution/views.py [hunk 1]
```diff
@@ -63,8 +63,11 @@ def get_queryset(self):
         queries.extend(
             query_utils.create_queries_by_lookup_field(self.request_data, [
                 'institution_name', 'editors_notes', 'institution_city', 'institution_country',
-                'resource', 'change_user', 'institution_id'
-            ])
+                'change_user', 'institution_id', 'resources', 'images'
+            ], search_fields_maps={'resources': ['resources__resource_name', 'resources__resource_details',
+                                                 'resources__resource_url'],
+                                   'images': ['images__image_filename', 'images__thumbnail',
+                                              'images__licence_details']})
```
Comment (philipkcl) [inline institution/views.py:70]: as your mention, we can query m2m field by `Model.objects.filter(resources__resource_name='aaa')`


is it Django also support m2m field in F function `F('resources__resource_name')`?

**Review by philipkcl — CHANGES_REQUESTED** [review]

File: core/management/commands/data_migration.py [hunk 2]
```diff
@@ -375,11 +379,24 @@ def _val_handler_upload__upload_status(row: dict, conn):
...
+
+    person_mappings[row['iperson_id']] = row['person_id']
+
+    return row
+
+
+def _val_handler_create_summary(row: dict, conn):
+    # CofkUnionPersonSummary links to primary key in CofkUnionPerson which is person_id
+    person_id = person_mappings[row['iperson_id']]
+    row['iperson_id'] = person_id
```
[the remark this answers is not in the document]
Comment (J4bbi) [reply core/management/commands/data_migration.py]: Resolved with https://github.com/culturesofknowledge/site-edit-2/pull/55/commits/454887b1f057388039a404db05c5cf42d075b972

File: core/management/commands/data_migration.py [hunk 3]
```diff
@@ -422,6 +439,40 @@ def _val_handler_work__catalogue(row: dict, conn):
...
+    if collect_person := CofkCollectPerson.objects.filter(iperson_id=row['iperson_id']).first():
+        row['iperson_id'] = collect_person.pk
+    print(row)
+
+    return row
+
+
+def _val_handler_union_person(row: dict, conn):
+    if union_person := CofkUnionPerson.objects.filter(iperson_id=row['union_iperson_id']).first():
+        row['union_iperson_id'] = union_person.pk
```
[the remark this answers is not in the document]
Comment (J4bbi) [reply core/management/commands/data_migration.py]: Resolved with https://github.com/culturesofknowledge/site-edit-2/pull/55/commits/3acb8189b54411c92bb642b2e702376273b8dbbe

File: person/templates/person/search_table_layout.html [hunk 4]
```diff
@@ -82,31 +85,38 @@
...
-            <td> 0 <!-- KTODO --> </td>
-            <td> 0 <!-- KTODO --> </td>
-            <td> 0 <!-- KTODO --> </td>
-            <td> <!-- KTODO  what is this field   CofkUnionPersonSummary.mentioned --> </td>
+            <td> {{ r.summary.sent }}</td>
+            <td> {{ r.summary.recd }} </td>
+            <td> {{ r.summary.all_works }} </td>
+            <td> {{ r.summary.mentioned }} </td>
+            <td>
+                {{ r.iperson_id }}</br>
```
[the remark this answers is not in the document]
Comment (J4bbi) [reply person/templates/person/search_table_layout.html]: Resolved https://github.com/culturesofknowledge/site-edit-2/pull/55/commits/d36e6778a7e06a8de3902791c90499cac927d884

File: core/management/commands/data_migration.py [hunk 5]
```diff
@@ -422,6 +439,40 @@ def _val_handler_work__catalogue(row: dict, conn):
...
+
+    return row
+
+
+def _val_handler_collect_work(row: dict, conn):
+    row['upload_status_id'] = row.pop('upload_status')
+
+    if union_work := CofkUnionWork.objects.filter(institution_id=row['repository_id']).first():
+        row['repository_id'] = union_work.pk
```
[the remark this answers is not in the document]
Comment (J4bbi) [reply core/management/commands/data_migration.py]: Resolved https://github.com/culturesofknowledge/site-edit-2/pull/55/commits/d36e6778a7e06a8de3902791c90499cac927d884

**Pull request merged**